packages/transformers/babel/src/babel7.js

```diff
@@ -82,6 +82,11 @@ export default async function babel7(
         remapAstLocations(babelCore.types, res.ast, map);
       }
     }
+    if (res.externalDependencies) {
+      for (let f of res.externalDependencies) {
+        asset.invalidateOnFileChange(f);
```

Comment:
And I guess Babel also doesn't differentiate between `onFileChange` and `onFileCreate`, so we'd have to do both (or check if the file exists)